ttnn/cpp/ttnn/operations/moreh/moreh_dot_op_backward/moreh_dot_backward.hpp

```diff
@@ -6,17 +6,28 @@
 #include "ttnn/decorators.hpp"
 namespace ttnn::operations::moreh::moreh_dot_backward {
 struct MorehDotBackward {
-    static std::vector<Tensor> invoke(
+    static std::vector<std::optional<Tensor>> invoke(
         const Tensor &output_grad,
         const Tensor &input,
         const Tensor &other,
-        std::optional<const Tensor> input_grad,
-        std::optional<const Tensor> other_grad,
+        const std::optional<const Tensor> &input_grad,
+        const std::optional<const Tensor> &other_grad,
+        const std::optional<MemoryConfig> &mem_config);
+
+    static std::vector<Tensor> create_async_output_tensors(
+        const std::vector<Tensor> &input_tensors, const std::vector<std::optional<Tensor>> &optional_inputs);
+
+    static std::vector<bool> create_async_tensor_flag(
+        const Tensor &output_grad,
+        const Tensor &input,
+        const Tensor &other,
+        const std::optional<const Tensor> &input_grad,
+        const std::optional<const Tensor> &other_grad,
         const std::optional<MemoryConfig> &mem_config);
 };
-}
+}  // namespace ttnn::operations::moreh::moreh_dot_backward
 
 namespace ttnn {
-constexpr auto moreh_dot_backward =
-    ttnn::register_operation<"ttnn::moreh_dot_backward", ttnn::operations::moreh::moreh_dot_backward::MorehDotBackward>();
+constexpr auto moreh_dot_backward = ttnn::
+    register_operation<"ttnn::moreh_dot_backward", ttnn::operations::moreh::moreh_dot_backward::MorehDotBackward>();
```

Comment:
```suggestion
    register_operation_with_auto_launch_op<"ttnn::moreh_dot_backward", ttnn::operations::moreh::moreh_dot_backward::MorehDotBackward>();
```

Reply:
Should be working now. I've run some tests and no error so far.